### PR TITLE
Fix #5357: Remove CDATA from translatable strings

### LINKED ISSUE #5357: [BUG]: CDATA causes issues with Translatewiki importing
### Describe the bug

As described in https://github.com/oppia/oppia-android/pull/5274/files#r1517940307 and observed in that PR, CDATA in strings causes one of three Translatewiki behaviors:
1. The string can be correctly translated.
2. The string is ignored (as is observed for privacy_policy_web_link, agree_to_terms, and terms_of_service_web_link.
3. The translation string is empty (#5274).

### Steps To Reproduce

See above.

### Expected Behavior

All strings should be translatable via Translatwiki.

### Screenshots/Videos

_No response_

### What device/emulator are you using?

_No response_

### Which Android version is your device/emulator running?

_No response_

### Which version of the Oppia Android app are you using?

_No response_

### Additional Context

I think this means we cannot use CDATA anymore and instead should prohibit it via a regex check. Existing CDATA strings will need to be escaped, instead (e.g. HTML tag markers will need to be replaced with '\&gt;' and '\&lt;').

### PR DESCRIPTION
**Pull request by adhiamboperes**
## Explanation

Fixes https://github.com/oppia/oppia-android/issues/5357

This PR includes the changes previously included in #5361, with an additional fix for missing closing html tags described in #5403:

```
faq_answer_app_language: Missing </strong> tag after the email address.
faq_answer_bug_reporting: Missing closing </ol> tag.
faq_answer_exploration_player: Missing closing </ul> tag.
faq_answer_update_app: Missing closing </ol> tag and closing </p> tag.
faq_answer_update_os: Missing closing </ol> tag and closing </p> tag.
```

This PR removes all CDATA declarations in the translatable strings.xml file and instead escapes all necessary characters: < and > (& didn't need to be escaped since no strings use that character at the moment). This is needed because Translatewiki doesn't seem to extract the HTML within the CDATA declaration correctly, so it may not be translated (some existing strings were never translated, and per https://github.com/oppia/oppia-android/pull/5274 the latest FAQ changes aren't being processed correctly (leading to empty translation strings being submitted).

This PR also introduces a regex check + test to ensure that CDATA isn't used anymore in strings.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -549,11 +549,11 @@
...
-  <string name="privacy_policy_web_link"><![CDATA[Please visit <a href="https://www.oppia.org/privacy-policy">this page</a> for the latest version of this privacy policy.]]></string>
+  <string name="privacy_policy_web_link">Please visit &lt;a href="https://www.oppia.org/privacy-policy"&gt;this page&lt;/a&gt; for the latest version of this privacy policy.</string>
   <!-- Terms of Service -->
   <string name="terms_of_service_title">Terms of Service</string>
-  <string name="agree_to_terms"><![CDATA[By using %s, you agree to our <br> <oppia-noninteractive-policy link="tos">Terms of Service</oppia-noninteractive-policy> and <oppia-noninteractive-policy link="privacy">Privacy Policy</oppia-noninteractive-policy>.]]></string>
-  <string name="terms_of_service_web_link"><![CDATA[Please visit <a href="https://www.oppia.org/terms">this page</a> for the latest version of these terms.]]></string>
+  <string name="agree_to_terms">By using %s, you agree to our &lt;br&gt; <oppia-noninteractive-policy link="tos">Terms of Service</oppia-noninteractive-policy> and <oppia-noninteractive-policy link="privacy">Privacy Policy</oppia-noninteractive-policy>.</string>
```
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]: Don't the custom tags also need to be escaped?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @adhiamboperes! The fixes look quite good. I just had one clarification comment to make sure everything was converted correctly.

That being said, I wonder if we need to perform any stronger analysis on the strings to ensure their HTML is well-formed? Or, perhaps that's something we can tackle as part of #4562? WDYT?

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (adhiamboperes) [reply app/src/main/res/values/strings.xml]: Fixed.

**Review by BenHenning — APPROVED** [review]
Thanks @adhiamboperes! This LGTM.

**Comment by BenHenning** [comment]
Going ahead and merging this since it is important to unblock Translatewiki. I believe we'll see the effects of this during the Thursday pull from Translatewiki.

**Pull request merged**